From 9387fca6765f74d01fa0d4439b32d988fd439549 Mon Sep 17 00:00:00 2001
From: Dmytro Lavrikov <dmytro.lavrikov@globallogic.com>
Date: Tue, 5 Nov 2019 17:55:41 +0200
Subject: [PATCH 3/4] CarSystemUi: Add launchDisplay attribute for
 CarFacetButton

This is intended to have the ability to choose display
on which activities will be launched when user clicks
facet button in navigation bar

Jira-ID: ADM-4110, ADM-3708
Signed-off-by: Dmytro Lavrikov <dmytro.lavrikov@globallogic.com>
Change-Id: I88f3a985ab33b2ef1e1c792cafb9b8fe9f5ade19
---
 .../src/com/android/systemui/statusbar/car/CarFacetButton.java   | 9 ++++++++-
 packages/SystemUI/res/values/attrs_car.xml                       | 2 ++
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/packages/CarSystemUI/src/com/android/systemui/statusbar/car/CarFacetButton.java b/packages/CarSystemUI/src/com/android/systemui/statusbar/car/CarFacetButton.java
index a371a1d..bce46ac7 100644
--- a/packages/CarSystemUI/src/com/android/systemui/statusbar/car/CarFacetButton.java
+++ b/packages/CarSystemUI/src/com/android/systemui/statusbar/car/CarFacetButton.java
@@ -16,6 +16,7 @@
 
 package com.android.systemui.statusbar.car;
 
+import android.app.ActivityOptions;
 import android.content.Context;
 import android.content.Intent;
 import android.content.res.TypedArray;
@@ -92,6 +93,7 @@ public class CarFacetButton extends LinearLayout {
         String longPressIntentString = typedArray.getString(R.styleable.CarFacetButton_longIntent);
         String categoryString = typedArray.getString(R.styleable.CarFacetButton_categories);
         String packageString = typedArray.getString(R.styleable.CarFacetButton_packages);
+        String displayString = typedArray.getString(R.styleable.CarFacetButton_launchDisplay);
         String componentNameString =
                 typedArray.getString(R.styleable.CarFacetButton_componentNames);
         try {
@@ -110,9 +112,14 @@ public class CarFacetButton extends LinearLayout {
                 mComponentNames = componentNameString.split(FACET_FILTER_DELIMITER);
             }
 
+            final ActivityOptions options = ActivityOptions.makeBasic();
+            if (displayString != null) {
+                options.setLaunchDisplayId(Integer.parseInt(displayString));
+            }
+
             setOnClickListener(v -> {
                 intent.putExtra(EXTRA_FACET_LAUNCH_PICKER, mSelected);
-                mContext.startActivityAsUser(intent, UserHandle.CURRENT);
+                mContext.startActivityAsUser(intent, options.toBundle(), UserHandle.CURRENT);
             });
 
             if (longPressIntentString != null) {
diff --git a/packages/SystemUI/res/values/attrs_car.xml b/packages/SystemUI/res/values/attrs_car.xml
index ced26c9..00497af 100644
--- a/packages/SystemUI/res/values/attrs_car.xml
+++ b/packages/SystemUI/res/values/attrs_car.xml
@@ -44,6 +44,8 @@
         <attr name="unselectedAlpha" />
         <!-- Render a "more" icon. Defaults true  -->
         <attr name="useMoreIcon" format="boolean" />
+        <!-- Set display on which activity will launch  -->
+        <attr name="launchDisplay" format="integer" />
 
     </declare-styleable>
 
-- 
2.7.4

